src/wallet/test/spend_tests.cpp

```diff
@@ -112,5 +112,38 @@ BOOST_FIXTURE_TEST_CASE(FillInputToWeightTest, BasicTestingSetup)
     // Note: We don't test the next boundary because of memory allocation constraints.
 }
 
+BOOST_FIXTURE_TEST_CASE(wallet_duplicated_preset_inputs_test, TestChain100Setup)
```

Comment:
yeah. I almost remove it but.. ended up leaving because (1) the severity of the bug and (2) because it's easier to debug and assert that the code is failing where is supposed to be failing (in the functional test case, as we use the `send` RPC command, we do have other stuff involved).

I'm fine either way, if this still doesn't convince you and more people don't see it useful, can remove it for sure.

Reply:
I also agree that the second unit test (which isn't really a unit test) seems redundant and also a little confusing to me. I think the unit test for `CoinsResult::Erase` in an earlier commit is correct for a unit test and the functional test is better suited for testing the wallet behavior